Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp

```diff
@@ -423,6 +461,18 @@ void NodeGraph::Save( IOStream & stream ) const
 			stream.Write( varAndHash.GetHash() );
 		}
 
+		// externally defined variables
+		const Array< AString > & definedStrings = BFFMacros::Get().Tokens();
+		const uint32_t definedStringsSize = static_cast<uint32_t>( definedStrings.GetSize() );
+		ASSERT( definedStringsSize == definedStrings.GetSize() );
+		stream.Write( definedStringsSize );
+		for ( uint32_t i = 0; i < definedStringsSize; ++i )
+		{
+			const AString & def = definedStrings[i];
+			stream.Write(def);
```

Comment:
It seems like this will cause all tokens from the previous run to be saved, including those that were defined in the config file itself. This will cause a permanent re-parsing situation.